### DISCUSSION TITLE
Visualiser not working: Can't load gluegen_rt.dll on Win 32 bits?

### DISCUSSION THREAD

**Discussion by mvackel** [winder/Universal-G-Code-Sender#2361] [General]
Hello,
I'm trying to use UGS in an old 32bits Windows computer but the Visualiser is not working. It doesn't show anything when I load a .nc file. It works perfectly on another 64bits computer.
After UGS starts, there is a error sign in the bottom status bar that clicking, displays a message saying that "couldn't load gluegen_rt".....  . It mentions this dll (gluegen_rt.dll) and a list of folders where it tried to load it. 

After a search, I could find this "gluegne_rt.dll" inside this JAR file:
UGS -> ugsplatform\modules\ext\com.willwinder.ugs-platform-ugslib\org-jogamp-gluegen\gluegen-rt-natives-windows-amd64.jar

I've unpacked it and copied to some of the folders mentioned in the error message (c:\ugs ,  c:\windows, c:\windows\system32) but the message doesn't change. It seems that this .dll is 64 bits.

So, please, how can I fix this? 

Thanks.

**Comment by breiler** [comment]
It seems that JOGL dropped the support for Windows 32-bit.
Your best option right now is to go back to UGS version 2.0.12 to get this working:
https://github.com/winder/Universal-G-Code-Sender/releases/tag/v2.0.12

I have asked them to reenable the 32-bit build again:
https://forum.jogamp.org/Reenable-windows-i586-td4043119.html

**Comment by phantom-pavlova** [comment]
my 32 bit (linux) antix distro has gluegen-rt (you have to look for it i think it is in libgluegen2??) but not libnativewindow_awt.so nor  libnativewindow_x11.so, they come from libjogl2-jni which is only available in 64 bit, which I could not build from source and the 2021 libs in the jogl archive just crash.
it is a shame, as I have a pile of 32bit laptops (but none in 64bit), that I can use to check the stuff that goes into the plotter, but i cant see the drawing.
the 64bit machine has to stay in the office, and as the plotter (grbl 1.1) is 15m x 3m, I cant move that. 
one thought is to use the usbip package to export the laptop usb connection to the office machine, then the laptop can ssh into the office machine and the xserver will let me use ugs - running in the office - on the laptop.
I am awaiting delivery of some wifi chips for the arduino, but still being able to see the plot and jog a bit whilst next to the machine makes life much less complicated.

**Reply by breiler** [reply]
Unfortunatley they have stopped distributing both the Java runtime and the graphics libraries we are using for Linux i586. You will have to use a 64-bit Linux to run Universal Gcode Sender.

**Reply by phantom-pavlova** [reply]
i will relay that to grbl. I have bought a pile of wifi chips to get around the issue ( if my 32 bit machine is not supported, I can talk directly between the 64 bit machine  and the 8 bit arduinos), but isn't that the point of java, that it runs on any machine??
- java - runs on your washing machine -
- java - download, all platforms -
- (i586 not supported) -

**Reply by phantom-pavlova** [reply]
who exactly are they?

**Reply by breiler** [reply]
Why would the GRBL project be interested in this? It has nothing to do with them.

I would like UGS to be able to run on old hardware. But we are relying on different projects that are now dropping support for older architectures so it isn't much we can do about it, unless we start to build and package those libraries ourselves. And that amount of work doesn't sound like fun to do for free...

> who exactly are they?

Java: https://en.wikipedia.org/wiki/Java_(programming_language)
JOGL: https://jogamp.org/

You have plenty of other options to try instead of UGS:
https://github.com/winder/Universal-G-Code-Sender/wiki/Software-comparison#gcode-sender
